week1/homework/src/server.js

```diff
@@ -10,6 +10,36 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    const requestUrl = request.url;
+    if (request.url === '/state') {
+      response.statusCode = 200;
+      response.setHeader('content-Type', 'application/json');
+      response.end(JSON.stringify({ state }));
+      // console.log(JSON.stringify({ state: state + 1 }));
+    }
+    else
+    if (request.url === '/add') {
+      response.statusCode = 200;
+      response.setHeader('content-Type', 'application/json');
+      response.end(JSON.stringify({ state: ++state }));
+    }
+    else
+    if (request.url === '/subtract') {
+      response.statusCode = 200;
+      response.setHeader('content-Type', 'application/json');
+      response.end(JSON.stringify({ state: --state }));
+    }
+    else
+    if (request.url === '/reset') {
+      response.statusCode = 200;
+      response.setHeader('content-Type', 'application/json');
+      response.end(JSON.stringify({ state: state = 10 }));
```

Comment:
You've written five times similar code: 
```
response.statusCode = 200;
response.setHeader('content-Type', 'application/json');
response.end(JSON.stringify({ state: state = 10 }));
```
The only thing changing in this code is the `statuscode` and the `state`. You should create a function if you are repeating the same code; the things that could be different in this block of code should be passed as arguments to that function. 
Refactor your code to make it more DRY (Don't Repeat Yourself)!